ws-server/src/server.rs

```diff
@@ -273,7 +300,10 @@ async fn background_task(
 						// complete batch response back to the client over `tx`.
 						let (tx_batch, mut rx_batch) = mpsc::unbounded::<String>();
 
-						for fut in batch.into_iter().filter_map(|req| methods.execute(&tx_batch, req, conn_id)) {
+						for fut in batch
```

Comment:
Random thought: it might be possible to calculate the overall sum of required resources for the whole batch call and accept/refuse it accordingly. Like it is, users could get themselves in a bind if half their batch is overquota and the calls are not idempotent. One could reply "you should have thought of that before using batch calls!", but that would be a bit unfriendly...

Might be good to have a test that illustrates the behaviour for batches in the presence of resource limiting.

Reply:
@maciejhirsz did we get a test&docs for this somewhere?

We had a conversation about this and came to the conclusion that while it's possible to calculate the batch "cost" up front, the semantics would become rather awkward and the overall behaviour of the server a bit hard to understand and reason about; on a busy server a batch could end up in a situation where it never gets enough resources to run or, conversely, a sequentially executed batch might block other requests from running.